pkg/storage/db_test.go

```diff
@@ -0,0 +1,374 @@
+package storage
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/alicebob/miniredis/v2"
+	"github.com/goccy/go-json"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func getDBImplementations(t *testing.T) []ServiceStorage {
+	boltDB := setupBoltDB(t)
+	redisDB := setupRedisDB(t)
+
+	dbImpls := make([]ServiceStorage, 0)
+	dbImpls = append(dbImpls, boltDB, redisDB)
+	return dbImpls
+}
+
+func setupBoltDB(t *testing.T) *BoltDB {
```

Comment:
what is the value in testing multiple DB implementations at the same time?

do you foresee us needing to run multiple DBs in a real environment?

Reply:
This guarantees that the same input, error messages, and output happen with any implementation of any db we decide to implement.